app/routes.py

```diff
@@ -1,2 +1,188 @@
+                tasks_list = Task.query.order_by(Task.title.desc()).all()
+        else:
+            tasks_list = Task.query.all()
+        response = []
+        for my_task in tasks_list:
+            response.append(my_task.to_dict())
+        return jsonify(response), 200
+
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"], strict_slashes=False)
+def get_task(task_id):
+    task_id = int(task_id)
+    my_task = Task.query.get(task_id)
+    if request.method == "GET":
+        if my_task is None:
+            return "", 404
+        else:
+            return make_response({"task":my_task.to_dict()}), 200
+
+    elif request.method == "PUT":
+        request_body = request.get_json()
+        try:
+            my_task.title = request_body["title"]
+            my_task.description = request_body["description"]
+            if "completed_at" in request_body :
+                my_task.completed_at = request_body["completed_at"]
+            db.session.commit()        
+            return make_response({"task":my_task.to_dict()}), 200
+        except AttributeError as ae:
+            return make_response("", 404)
+
+    elif request.method == "DELETE":
+        if my_task is None:
+            return "", 404
+        else:        
+            db.session.delete(my_task)
+            db.session.commit()
+            myResponse = 'Task ' + str(task_id) + ' "' + my_task.title + '" successfully deleted'
+            return make_response({"details":myResponse}), 200
+
+@tasks_bp.route("/<task_id>/<completed_status>", methods=["PATCH"], strict_slashes=False )
+def mark_complete(task_id, completed_status):
+    task_id = int(task_id)
+    my_task = Task.query.get(task_id)
+    if request.method == "PATCH":
+        if my_task is None:
+            return "", 404
+        else:
+            if completed_status == "mark_complete":
+                my_task.completed_at = datetime.now()
+                send_slack_notice(my_task.title)
+            elif completed_status == "mark_incomplete":
+                my_task.completed_at = None
+            db.session.commit() 
+            return make_response({"task" : my_task.to_dict()}), 200
+
+
+def send_slack_notice(this_task_title):
+    #here we need to create an http.request to the slack end point
+    myNotice = urllib.parse.quote_plus("Someone just completed the task " + this_task_title)
+    my_url = "https://slack.com/api/chat.postMessage?channel=task-notifications&text=" + myNotice + "&pretty=1"
+    #add a request.header "Authorization", "super secret"
+    my_token = os.environ.get("MY_SLACK_TOKEN")
+    my_headers = {"Authorization": f"Bearer {my_token}"}
+    #send
+    r = requests.post(my_url, data="", headers=my_headers)
+
+@goals_bp.route("", methods=["GET", "POST"], strict_slashes=False)
+def create_goal():
+    if request.method == "POST":
+        request_body = request.get_json()
+        if "title" not in request_body:
+            return {"details": "Invalid data"}, 400
```

Comment:
Maybe respond with more detail, like 

```suggestion
            return {"details": "'title' is required"}, 400
```